src/scicloj/02_numpy.clj

```diff
@@ -439,17 +858,59 @@ plt.ylabel('number'));
 ^k/dataset
 (ds/head csv-data 3)
 
+(tech.v3.datatype.functional/mean (dtype/->reader (-> csv-data last last)))
+
+(letfn [(mean [x] (float (/ (apply + x) (count x))))]
+  (nd-aggregate mean
+                (dtype/->reader (-> csv-data last last))))
+
+
 ["Computation on Arrays: Broadcasting
  ------------------------------------------------"]
 
-["### WIP"]
-
 ["Comparisons, Masks, and Boolean Logic
  ------------------------------------------------"]
 
+(defn mask [f t]
+  (-> (dtype/emap #(if (f %) 1 0) :int32 t)
+      dtt/->tensor
+      ravel))
+
+(mask zero? (eye 5))
+
 ["Fancy Indexing
  ------------------------------------------------"]
```

Comment:
I guess fancy indexing is exactly what `dtt/select` does, no?

Reply:
Though perhaps not, since fancy indexing can also define the output shape, not just extract a sub-shape.

Reply:
Good question.  I'm not sure what that meant, so I left it alone!